powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/util/EncryptionUtil.java

```diff
@@ -54,4 +84,25 @@ public static void processEncryptedResponse(StepContext<?, EciesEncryptedRespons
                 decryptedMessage
         );
     }
+
+    /**
+     * Derive associated data for ECIES.
+     * @param eciesScope ECIES scope.
+     * @param version Crypto protocol version.
+     * @param applicationKey Application key.
+     * @param activationId Activation identifier.
+     * @return Associated data.
+     */
+    public static byte[] deriveAssociatedData(EciesScope eciesScope, PowerAuthVersion version, String applicationKey, String activationId) {
+        if (version.useTimestamp()) {
+            if (eciesScope == EciesScope.ACTIVATION_SCOPE) {
+                return ByteUtils.concatStrings(version.value(), applicationKey, activationId);
+            } else {
+                return ByteUtils.concatStrings(version.value(), applicationKey);
+            }
+        } else {
+            return null;
+        }
+    }
+
```

Comment:
This should be a part of `powerauth-crypto` library. We have the same code duplicated in the server's implelementation.

Reply:
Ah, yes, I wanted to move this class.